From 31d62ec9232155c04f6aa1405025d4fddd602caf Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Wed, 16 Feb 2011 19:56:00 -0200
Subject: [PATCH 1/2] Add SUPPORTED_* constants to pxe_api.h

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <1297886160-26299-1-git-send-email-ehabkost@redhat.com>
Patchwork-id: 18351
O-Subject: [RHEL-6 gpxe PATCH] Add SUPPORTED_* constants to pxe_api.h
Bugzilla: 661840
RH-Acked-by: Gleb Natapov <gleb@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>

This needs to be applied before the UNDI polling patch I have submitted
previously (that was already ACKed).

Part of upstream commit:

commit 0b922595fe770e86a761339ff57ddd7ac12a50d2
Author: Michael Brown <mcb30@etherboot.org>
Date:   Tue Jun 23 22:42:55 2009 +0100

    [pxe] Fix interoperability with the Intel DOS UNDI driver

    The Intel DOS UNDI driver fails when run on top of gPXE because we do
    not fill in the ServiceFlags field in PXENV_UNDI_GET_IFACE_INFO.

    Fix by filling in the ServiceFlags field with reasonable values
    indicating our approximate feature capabilities.

I have not pulled the part that actually set the flags on
undi_get_iface_info, on pxe_undi.c, to reduce risk.

Bugzilla: 668005

Signed-off-by: Eduardo Habkost <ehabkost@raisama.net>
---
 src/arch/i386/include/pxe_api.h |   19 +++++++++++++++++++
 1 files changed, 19 insertions(+), 0 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@raisama.net>
---
 src/arch/i386/include/pxe_api.h |   19 +++++++++++++++++++
 1 files changed, 19 insertions(+), 0 deletions(-)

diff --git a/src/arch/i386/include/pxe_api.h b/src/arch/i386/include/pxe_api.h
index b3d4bca..bfdc94d 100644
--- a/src/arch/i386/include/pxe_api.h
+++ b/src/arch/i386/include/pxe_api.h
@@ -1415,6 +1415,25 @@ extern PXENV_EXIT_t pxenv_undi_get_nic_type (
 /** PXE API function code for pxenv_undi_get_iface_info() */
 #define	PXENV_UNDI_GET_IFACE_INFO	0x0013
 
+/** Broadcast supported */
+#define SUPPORTED_BROADCAST		0x0001
+/** Multicast supported */
+#define SUPPORTED_MULTICAST		0x0002
+/** Functional/group addressing supported */
+#define SUPPORTED_GROUP			0x0004
+/** Promiscuous mode supported */
+#define SUPPORTED_PROMISCUOUS		0x0008
+/** Software settable station address */
+#define SUPPORTED_SET_STATION_ADDRESS	0x0010
+/** InitiateDiagnostics supported */
+#define SUPPORTED_DIAGNOSTICS		0x0040
+/** Reset MAC supported */
+#define SUPPORTED_RESET			0x0400
+/** Open / Close Adapter supported */
+#define SUPPORTED_OPEN_CLOSE		0x0800
+/** Interrupt Request supported */
+#define SUPPORTED_IRQ			0x1000
+
 /** Parameter block for pxenv_undi_get_iface_info() */
 struct s_PXENV_UNDI_GET_IFACE_INFO {
 	PXENV_STATUS_t	Status;		/**< PXE status code */
-- 
1.7.3.2

